### PR TITLE
Add and harden various CMake options

### LINKED ISSUE #114: uncrustify running as part of normal build process?
Mike Collette reports uncrustify running during a normal build of Axom, which is causing it to fail. 

That was a surprise to me, I thought style checks were opt in via specific targets. 

Working on getting more info from Mike.

### LINKED ISSUE #560: Add a config option to guard against building axom's tools
A user requested adding a guard for axom to not build its tools like `mesh_tester` and `convert_sidre_protocol` (in `axom/src/tools`).

Should this use an existing flag, like `AXOM_ENABLE_EXAMPLES` or a new one like `AXOM_ENABLE_TOOLS` ?

### PR DESCRIPTION
**Pull request by white238**
- This PR is a bugfix/feature
- It does the following (modify list as needed):
  - Adds new AXOM_ENABLE_TOOLS and guards the appropriate tools from built
  - Harden our guards against developer tools from bleeding into users

Closes #560 
Closes #114

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -12,9 +12,6 @@ else()
     cmake_minimum_required(VERSION 3.8.2)
 endif()
 
-message(STATUS "Using CMake version ${CMAKE_VERSION}")
```
Comment (white238) [inline src/CMakeLists.txt:15]: This is output by BLT

**Review by cyrush — APPROVED** [review]

**Review by rhornung67 — APPROVED** [review]

**Review by kennyweiss — APPROVED** [review]
Thanks @white238 !

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -50,9 +47,31 @@ if (“${PROJECT_SOURCE_DIR}” STREQUAL “${CMAKE_SOURCE_DIR}”)
...
+    # unless an explicit executable path is given
+    set(AXOM_USED_BLT_TOOLS
+        CLANGFORMAT
+        CPPCHECK
+        DOXYGEN
+        SPHINX)
+    foreach(_tool ${AXOM_NOT_USED_BLT_TOOLS})
+        if(NOT ${_tool}_EXECUTABLE)
+            set(ENABLE_${_tool} OFF CACHE BOOL "")
+        endif()
```
Comment (agcapps) [inline src/CMakeLists.txt:73]: Do we need the other leg of the `if`?  So, if the user has defined `DOXYGEN_EXECUTABLE`, do we want to have `set(ENABLE_${_tool} ON CACHE BOOL "")` or is that not needed?

**Review by agcapps — APPROVED** [review]
Thanks Chris, just the one question.

File: src/CMakeLists.txt (hunk 2 above)
Comment (white238) [reply src/CMakeLists.txt:75]: Don't need to but will guard against when i possibly turn these off by default in BLT.

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -18,6 +18,16 @@ The format of this file is based on [Keep a Changelog](http://keepachangelog.com
...
 
 
+## [Unreleased] - Release date yyyy-mm-dd
+
+### Added
+- Added new CMake option to allow users to turn off Axom created tools: `AXOM_ENABLE_TOOLS`
+
+### Changed
+- `MFEMSidreDataCollection` now reuses FESpace/QSpace objects with the same basis
+- Harden BLT tools (style, code quality, etc.) against being enabled for users.  Developers will
```
Comment (kennyweiss) [inline RELEASE-NOTES.md:28]:
```suggestion
- Harden configuration options for BLT tools (style, code quality, etc.) against accidentally being enabled for users.  Developers will
```

**Review by kennyweiss — APPROVED** [review]

**Pull request merged**